src/formatDate.js

```diff
@@ -51,6 +51,32 @@
 
 function formatDate(date, fromFormat, toFormat) {
   // write code here
+  let newDate = '';
+  const dateArray = date.split(fromFormat[3]);
+  const dateObject = {};
+
+  for (let i = 0; i < 3; i++) {
+    dateObject[fromFormat[i]] = dateArray[i];
+  }
+
+  for (let i = 0; i < 3; i++) {
+    const index = toFormat.indexOf(fromFormat[i]);
+
+    if (index === -1) {
+      if (fromFormat[i].length === 2) {
+        dateObject['YYYY']
+          = Number(dateArray[i]) < 30 ? 20 + dateArray[i] : 19 + dateArray[i];
+      }
+
+      if (fromFormat[i].length === 4) {
+        dateObject['YY'] = dateArray[i].slice(2);
+      }
+    }
+
+    newDate += dateObject[toFormat[i]] + toFormat[3];
+  }
```

Comment:
fromFormat[i] - is repeating several times. You can create seperate variable for it, for example const oldFormat = fromFormat[i] and than use it.
Better check on string contain instead length
if (fromFormat[i] === 'YYYY')